src/languages/yaml.js

```diff
@@ -102,7 +102,7 @@ export default function(hljs) {
       // Indentation of subsequent lines must be the same to
       // be considered part of the block
       className: 'string',
-      begin: '[\\|>]([0-9]?[+-])?[ ]*\\n( *)[\\S ]+\\n(\\2[\\S ]+\\n?)*'
+      begin: '[\\|>]([0-9]?[+-])?[ ]*\\n( *)[^\\n]+\\n(\\2[^\\n]+\\n?)*'
```

Comment:
```suggestion
      begin: '[\\|>]([0-9]?[+-])?[ ]*\\n( +)[^\\n]+\\n(\\2[^\\n]+\\n?)*'
```

While we're talking about the behavior of this regex in this PR, should it be one or more spaces for the indentation? Having no indentation is invalid.

```yaml
key: |
multi-string
value
```

Reply:
In this case (and what I know of YAML) I'd agree but I'm not certain what's going on with all the other `[\\|>]([0-9]?[+-])?` there... are there some other variants we're not considering?  I think the 0-9 is to allow offsets/extra indents but I'm not sure what `\\` and `>`... is the \` is just needless escaping then that leaves only the `>`... does that change anything?

Reply:
My favorite resource for deciphering YAML multi-line values... https://yaml-multiline.info/

- You can use 1-9 for indentation if the first line's indentation won't match the rest of the lines

According to yamllint.com, the `\` is valid,

this:

```yaml
key: \
  multi-string
  value
```

becomes this:

```yaml
key: "\\ multi-string value"
```